tutorials/appengine-ruby-rails-activejob-pubsub.md

```diff
@@ -0,0 +1,272 @@
+to generate a cost estimate based on your projected usage. New Cloud Platform users might be eligible for a
+[free trial](https://cloud.google.com/free-trial).
+
+## Creating your background job
+
+Starting with version 4.2, Ruby on Rails includes an abstraction layer around background job processing called
+[ActiveJob](http://guides.rubyonrails.org/active_job_basics.html). This abstraction allows you to write the queuing
+and execution logic independently of queue and job runner implementations.
+
+Ruby on Rails provides command-line tools for generating templated skeletons for things such as database migrations,
+controllers, and even background jobs.
+
+You will create a job named `HelloJob` that will accept a `name` argument and "Hello #{name}" to standard output.
+
+1. Use the Rails generator feature to create `HelloJob`:
+
+        bin/rails generate job Hello
+
+   Rails creates stub files from templates:
+
+        invoke  test_unit
+        create    test/jobs/hello_job_test.rb
+        create  app/jobs/hello_job.rb
+        create  app/jobs/application_job.rb
+
+1. Edit your `app/jobs/hello_job.rb` with the following:
+
+        class HelloJob < ApplicationJob
+          queue_as :default
+
+          def perform(name)
+            # Do something later
+            puts "Hello, #{name} from ActiveJob!"
+          end
+        end
+
+## Create a test url to queue the job
+
+You will create a controller named `HelloController` that will provide an action called `say` which will queue
+our `HelloJob` to execute in the background.
+
+1. Use the Rails generator feature to create `HelloController`:
+
+        bin/rails generate controller Hello
+
+   Rails creates stub files from templates:
+
+        create  app/controllers/hello_controller.rb
+        invoke  erb
+        create    app/views/hello
+        invoke  test_unit
+        create    test/controllers/hello_controller_test.rb
+        invoke  helper
+        create    app/helpers/hello_helper.rb
+        invoke    test_unit
+        invoke  assets
+        invoke    coffee
+        create      app/assets/javascripts/hello.coffee
+        invoke    scss
+        create      app/assets/stylesheets/hello.scss
+
+1. Add a `say` action to `HelloController`. Edit your `app/controllers/hello_controller.rb` with the following:
+
+        class HelloController < ApplicationController
+          def say
+            HelloJob.perform_later(params[:name])
+            render plain: 'OK'
+          end
+        end
+
+   This action will queue our `HelloJob` with the provided request parameter `name`.
+
+1. Create a route to this action. In `config/routes.rb`, add:
+
+        get '/hello/:name', to: 'hello#say'
+
+   When you make an HTTP GET request to `/hello/Jeff`, the `HelloController` will handle the request using the `say`
+   action with parameter `:name` as "Jeff"
+
+1. Add a route for Google App Engine health checks. In `config/routes.rb`, add:
```

Comment:
This isn't necessary.

The health check only fails on 5xx response codes.  By default, Rails will return a 404 which the health check treats as a success.